### ISSUE TITLE
Discriminator-based inheritance only shows single level

### ISSUE THREAD

**Issue by weirdan** [Redocly/redoc#368]
Consider the following type hierarchy:
```json
  "definitions": {
    "Base": {
      "type": "object",
      "discriminator": "type",
      "required": [ "type" ],
      "properties": {
        "type": {
          "type": "string",
          "enum": [ "Base", "Intermediate", "Leaf" ]
        }
      }
    },
    "Intermediate": {
      "allOf": [
        { "$ref": "#/definitions/Base" },
        {
          "properties": {
            "type": {
              "type": "string",
              "enum": [ "Intermediate", "Leaf" ]
            }
          }
        }
      ]
    },
    "Leaf": {
      "allOf": [
        { "$ref": "#/definitions/Intermediate" },
        {
          "properties": {
            "type": {
              "type": "string",
              "enum": [ "Leaf" ]
            }
          }
        }
      ]
    }
  }
```
Currently, wherever `Base` is referenced, ReDoc shows that either `Base` or `Intermediate` could be provided, but it doesn't show that `Leaf` could also be used there, as it "inherits" from `Intermediate` and `Base` type enum allows it.

[Live demo](https://rebilly.github.io/ReDoc/?url=https://gist.githubusercontent.com/weirdan/9d842459f6f9c7790713e5c1962db1fc/raw/268ec0c282ad5a9be4c5afdd959d860b265c65e4/types.json#)

[swagger.json](https://gist.githubusercontent.com/weirdan/9d842459f6f9c7790713e5c1962db1fc/raw/268ec0c282ad5a9be4c5afdd959d860b265c65e4/types.json)

**Comment by RomanGotsiy** [comment]
@weirdan hopefully this will be addressed in 2.0.
Thanks for the great bug report!

**Comment by RomanHotsiy** [comment]
This is a dup of #210
